### PR TITLE
Handle BOMs in file contents

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

**Comment by ChadNedzlek** [comment]
Oh BOM's, so sad.

File: src/Microsoft.DotNet.Darc/src/DarcLib/IGitRepoExtension.cs [hunk 1]
```diff
@@ -19,7 +20,11 @@ public static string GetDecodedContent(this IGitRepo gitRepo, string encodedCont
             try
             {
                 byte[] content = Convert.FromBase64String(encodedContent);
-                return Encoding.UTF8.GetString(content);
+                using (var str = new MemoryStream(content, false))
+                using (var reader = new StreamReader(str))
```
Comment (mmitche) [inline src/Microsoft.DotNet.Darc/src/DarcLib/IGitRepoExtension.cs]: Can you put a little note in here as to why this was implemented this way vs. the more obvious UTF8.GetString?

Comment (alexperovich) [reply src/Microsoft.DotNet.Darc/src/DarcLib/IGitRepoExtension.cs]: sure

Comment (ChadNedzlek) [reply src/Microsoft.DotNet.Darc/src/DarcLib/IGitRepoExtension.cs]: I hope no one uses some other none UTF8 encoding. :-) I don't think I'd mind removing the BOM's? We should only be doing this to file that we "control", right?

Comment (mmitche) [reply src/Microsoft.DotNet.Darc/src/DarcLib/IGitRepoExtension.cs]: yeah, we 'control' it.  I think removing the BOM's is the right thing.  I hate BOMs

**Pull request merged**